diff --git a/src/hb-algs.hh b/src/hb-algs.hh
index bc170b0..5ddd3b1 100644
--- a/src/hb-algs.hh
+++ b/src/hb-algs.hh
@@ -544,14 +544,12 @@ static inline unsigned int
 hb_popcount (T v)
 {
 #if (defined(__GNUC__) && (__GNUC__ >= 4)) || defined(__clang__)
-  if (sizeof (T) <= sizeof (unsigned int))
-    return __builtin_popcount (v);
-
-  if (sizeof (T) <= sizeof (unsigned long))
-    return __builtin_popcountl (v);
-
-  if (sizeof (T) <= sizeof (unsigned long long))
-    return __builtin_popcountll (v);
+  if (sizeof (T) <= 8) {
+    uint64_t r1;
+    uint64_t r2 = v;
+    asm("POPCNT %0, %1" : "=r" (r1) : "r" (r2));
+    return r1;
+  }
 #endif
 
   if (sizeof (T) <= 4)
