diff --git a/libass/ass_shaper.c b/libass/ass_shaper.c
index 3718e45..acefd7c 100644
--- a/libass/ass_shaper.c
+++ b/libass/ass_shaper.c
@@ -360,26 +360,6 @@ cached_v_origin(hb_font_t *font, void *font_data, hb_codepoint_t glyph,
     return true;
 }
 
-static hb_position_t
-get_h_kerning(hb_font_t *font, void *font_data, hb_codepoint_t first,
-                 hb_codepoint_t second, void *user_data)
-{
-    FT_Face face = font_data;
-    FT_Vector kern;
-
-    if (FT_Get_Kerning(face, first, second, FT_KERNING_DEFAULT, &kern))
-        return 0;
-
-    return kern.x;
-}
-
-static hb_position_t
-get_v_kerning(hb_font_t *font, void *font_data, hb_codepoint_t first,
-                 hb_codepoint_t second, void *user_data)
-{
-    return 0;
-}
-
 static hb_bool_t
 cached_extents(hb_font_t *font, void *font_data, hb_codepoint_t glyph,
                hb_glyph_extents_t *extents, void *user_data)
@@ -476,10 +456,6 @@ static hb_font_t *get_hb_font(ASS_Shaper *shaper, GlyphInfo *info)
                 metrics, NULL);
         hb_font_funcs_set_glyph_v_origin_func(funcs, cached_v_origin,
                 metrics, NULL);
-        hb_font_funcs_set_glyph_h_kerning_func(funcs, get_h_kerning,
-                metrics, NULL);
-        hb_font_funcs_set_glyph_v_kerning_func(funcs, get_v_kerning,
-                metrics, NULL);
         hb_font_funcs_set_glyph_extents_func(funcs, cached_extents,
                 metrics, NULL);
         hb_font_funcs_set_glyph_contour_point_func(funcs, get_contour_point,
